<h1><strong>Data preprocessing</strong></h1>

<h3>Team Science Bros</h3>

In this notebook file you shall find data preprocessing script.

In [ ]:
# Importing pandas
import pandas as pd

In [ ]:
# Importing and reading the dataset as df 
df = pd.read_csv('/content/Raw_Stroke_Dataset.csv')
df

id  gender   age  hypertension  heart_disease ever_married  \
0      9046    Male  67.0             0              1          Yes   
1     51676  Female  61.0             0              0          Yes   
2     31112    Male  80.0             0              1          Yes   
3     60182  Female  49.0             0              0          Yes   
4      1665  Female  79.0             1              0          Yes   
...     ...     ...   ...           ...            ...          ...   
5105  18234  Female  80.0             1              0          Yes   
5106  44873  Female  81.0             0              0          Yes   
5107  19723  Female  35.0             0              0          Yes   
5108  37544    Male  51.0             0              0          Yes   
5109  44679  Female  44.0             0              0          Yes   

          work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0           Private          Urban             228.69  36.6  formerly smoked   
1     Self-employed          Rural             202.21   NaN     never smoked   
2           Private          Rural             105.92  32.5     never smoked   
3           Private          Urban             171.23  34.4           smokes   
4     Self-employed          Rural             174.12  24.0     never smoked   
...             ...            ...                ...   ...              ...   
5105        Private          Urban              83.75   NaN     never smoked   
5106  Self-employed          Urban             125.20  40.0     never smoked   
5107  Self-employed          Rural              82.99  30.6     never smoked   
5108        Private          Rural             166.29  25.6  formerly smoked   
5109       Govt_job          Urban              85.28  26.2          Unknown   

      stroke  
0          1  
1          1  
2          1  
3          1  
4          1  
...      ...  
5105       0  
5106       0  
5107       0  
5108       0  
5109       0  

[5110 rows x 12 columns]

In [ ]:
# Caluclating the sum of missing values 
df.isnull().sum()

id                     0
gender                 0
age                    0
hypertension           0
heart_disease          0
ever_married           0
work_type              0
Residence_type         0
avg_glucose_level      0
bmi                  201
smoking_status         0
stroke                 0
dtype: int64

<strong>We have learnt that the 'bmi' attribute has 201 missing values.</strong><br>
We need to resolve this issue before we go any further.

In [ ]:
# Dropping the null values in 'BMI' coulmn
df.dropna(subset=['bmi'],inplace = True)
df

id  gender   age  hypertension  heart_disease ever_married  \
0      9046    Male  67.0             0              1          Yes   
2     31112    Male  80.0             0              1          Yes   
3     60182  Female  49.0             0              0          Yes   
4      1665  Female  79.0             1              0          Yes   
5     56669    Male  81.0             0              0          Yes   
...     ...     ...   ...           ...            ...          ...   
5104  14180  Female  13.0             0              0           No   
5106  44873  Female  81.0             0              0          Yes   
5107  19723  Female  35.0             0              0          Yes   
5108  37544    Male  51.0             0              0          Yes   
5109  44679  Female  44.0             0              0          Yes   

          work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0           Private          Urban             228.69  36.6  formerly smoked   
2           Private          Rural             105.92  32.5     never smoked   
3           Private          Urban             171.23  34.4           smokes   
4     Self-employed          Rural             174.12  24.0     never smoked   
5           Private          Urban             186.21  29.0  formerly smoked   
...             ...            ...                ...   ...              ...   
5104       children          Rural             103.08  18.6          Unknown   
5106  Self-employed          Urban             125.20  40.0     never smoked   
5107  Self-employed          Rural              82.99  30.6     never smoked   
5108        Private          Rural             166.29  25.6  formerly smoked   
5109       Govt_job          Urban              85.28  26.2          Unknown   

      stroke  
0          1  
2          1  
3          1  
4          1  
5          1  
...      ...  
5104       0  
5106       0  
5107       0  
5108       0  
5109       0  

[4909 rows x 12 columns]

<strong>Now, that we have dropped the missing values, we check again.</strong>

In [ ]:
# Making sure that dataset has no missing data
df.isnull().sum()

id                   0
gender               0
age                  0
hypertension         0
heart_disease        0
ever_married         0
work_type            0
Residence_type       0
avg_glucose_level    0
bmi                  0
smoking_status       0
stroke               0
dtype: int64

<strong>Label Encoding</strong>

In [ ]:
# Import preprocessing from sklearn for Label Encoding
from sklearn import preprocessing
label_encoder = preprocessing.LabelEncoder()


# Encode labels in column 'gender'.
# Male : 1 , Female : 0
df['gender']= label_encoder.fit_transform(df['gender'])


# Encode labels in column 'ever_married'.
# Yes : 1 , No : 0
df['ever_married']= label_encoder.fit_transform(df['ever_married'])


# Encode labels in column 'Residence_type'.
# Urban : 1, Rural : 0
df['Residence_type']= label_encoder.fit_transform(df['Residence_type'])


# Encode labels in column 'work_type'.
# Govt_job : 0, Never_worked : 1,Private : 2,Self-employed : 3  children : 4, 
df['work_type']= label_encoder.fit_transform(df['work_type'])


# Encode labels in column 'smoking_status'.
# Unknown : 0, formerly smoked : 1, never smoked : 2, smokes : 3
df['smoking_status']= label_encoder.fit_transform(df['smoking_status'])


df

id  gender   age  hypertension  heart_disease  ever_married  \
0      9046       1  67.0             0              1             1   
2     31112       1  80.0             0              1             1   
3     60182       0  49.0             0              0             1   
4      1665       0  79.0             1              0             1   
5     56669       1  81.0             0              0             1   
...     ...     ...   ...           ...            ...           ...   
5104  14180       0  13.0             0              0             0   
5106  44873       0  81.0             0              0             1   
5107  19723       0  35.0             0              0             1   
5108  37544       1  51.0             0              0             1   
5109  44679       0  44.0             0              0             1   

      work_type  Residence_type  avg_glucose_level   bmi  smoking_status  \
0             2               1             228.69  36.6               1   
2             2               0             105.92  32.5               2   
3             2               1             171.23  34.4               3   
4             3               0             174.12  24.0               2   
5             2               1             186.21  29.0               1   
...         ...             ...                ...   ...             ...   
5104          4               0             103.08  18.6               0   
5106          3               1             125.20  40.0               2   
5107          3               0              82.99  30.6               2   
5108          2               0             166.29  25.6               1   
5109          0               1              85.28  26.2               0   

      stroke  
0          1  
2          1  
3          1  
4          1  
5          1  
...      ...  
5104       0  
5106       0  
5107       0  
5108       0  
5109       0  

[4909 rows x 12 columns]

<strong>Now, we shall be dropping the 'id' attribute as we are not using it in modeling stage.</strong>

In [ ]:
# Dropping the 'id' attribute of df
df.drop(['id'], axis=1)

gender   age  hypertension  heart_disease  ever_married  work_type  \
0          1  67.0             0              1             1          2   
2          1  80.0             0              1             1          2   
3          0  49.0             0              0             1          2   
4          0  79.0             1              0             1          3   
5          1  81.0             0              0             1          2   
...      ...   ...           ...            ...           ...        ...   
5104       0  13.0             0              0             0          4   
5106       0  81.0             0              0             1          3   
5107       0  35.0             0              0             1          3   
5108       1  51.0             0              0             1          2   
5109       0  44.0             0              0             1          0   

      Residence_type  avg_glucose_level   bmi  smoking_status  stroke  
0                  1             228.69  36.6               1       1  
2                  0             105.92  32.5               2       1  
3                  1             171.23  34.4               3       1  
4                  0             174.12  24.0               2       1  
5                  1             186.21  29.0               1       1  
...              ...                ...   ...             ...     ...  
5104               0             103.08  18.6               0       0  
5106               1             125.20  40.0               2       0  
5107               0              82.99  30.6               2       0  
5108               0             166.29  25.6               1       0  
5109               1              85.28  26.2               0       0  

[4909 rows x 11 columns]